mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -5787,17 +6615,23 @@ struct XdlopsGemmV2RewritePattern
 
       auto destOffsetA = ilmkb.create<AddIOp>(loc, ilmkiv, kOffsetA);
 
-      auto valueA =
-          ilmkb.create<memref::LoadOp>(loc, dataType, op.matrixA(), sourceOffsetA);
+      Value valueA;
+      if (KPack > 1) {
+        valueA = emitLoadLogic(
+            ilmkb, loc, op.matrixA().getType().template cast<MemRefType>(),
```

Comment:
Couldn't this always be emitLoadLogic and it'll drop down to the memref.load in the scalar case?

Reply:
I guess it's possible. But I don't want to change this until a full passing CI. I'd do it in a later PR to further clean this up.